### PR TITLE
[#SELC-1473] Add developer portal due to external api

### PR DESCRIPTION
**Pull request by iwoak**
### List of changes
- Added static asset due to api readable page generated by redoc to storage
- Added page for each environment to explain external api to storage

### Motivation and context
We would create simple doc page using openAPI for users/clients of self care rest API

### Type of changes
- [X] Add new resources
- [X] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [X] DEV
- [X] UAT
- [X] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [X] Yes
- [] No

### Other information

---

### If PR is partially applied, why? (reserved to mantainers)
There aren't blocking cause, we applied now just on DEV from that branch, after approving and success test we'll apply it on other envs

### CODE REVIEW

File: src/core/developer.tf [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  depends_on = [module.apim_external_api_ms_v1]
+
+  provisioner "local-exec" {
+    command = <<EOT
+      az rest \
+        --method get \
+        --url https://management.azure.com/subscriptions/${data.azurerm_subscription.current.subscription_id}/resourceGroups/${local.resource_groups_name}/providers/Microsoft.ApiManagement/service/${local.service_name}/apis/${local.api_id} \
+        --url-parameters api-version=${local.azure_apim_api_version} export=true format=openapi \
+ 	--output-file ./env/${var.env}/developer/external/ms-external-api-v1.yaml
```
Comment (antonioT90) [inline src/core/developer.tf:21]: this file is not a gitignored destination, thus this invocation would brought to a file to commit that if runned locally could be confusing.

I suggest to use the location "${path.module}/.terraform/tmp/" as base path

File: src/core/developer.tf [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    command = <<EOT
+              az storage blob sync \
+                --container '$web' \
+                --account-name ${replace(replace(module.checkout_cdn.name, "-cdn-endpoint", "-sa"), "-", "")} \
+                --account-key ${module.checkout_cdn.storage_primary_access_key} \
+                --source "./env/${var.env}/developer/external/ms-external-api-v1.yaml" \
+                --destination 'developer/external/ms-external-api-v1.yaml'
+              az cdn endpoint purge \
+                -g ${azurerm_resource_group.checkout_fe_rg.name} \
+                -n ${module.checkout_cdn.name} \
```
Comment (antonioT90) [inline src/core/developer.tf:43]: use the extended name for these argument names

File: src/core/developer.tf [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    command = <<EOT
+              az storage blob sync \
+                --container '$web' \
+                --account-name ${replace(replace(module.checkout_cdn.name, "-cdn-endpoint", "-sa"), "-", "")} \
+                --account-key ${module.checkout_cdn.storage_primary_access_key} \
+                --source "./env/${var.env}/developer/external/index.html" \
+                --destination 'developer/external/index.html'
+              az cdn endpoint purge \
+                -g ${azurerm_resource_group.checkout_fe_rg.name} \
+                -n ${module.checkout_cdn.name} \
```
Comment (antonioT90) [inline src/core/developer.tf:66]: use the extended name for these argument names

File: src/core/developer.tf (hunk 1 above)
Comment (rodoviti) [reply src/core/developer.tf:21]: @antonioT90 the _--output-file_ parameter fails if the destination directory does not exist...any suggestions?

**Review by antonioT90 — APPROVED** [review]

**Pull request merged**